__tests__/endpoint.test.js

```diff
@@ -3,21 +3,48 @@ const data = require("../db/data/test-data");
 const db = require("../db/connection");
 const seed = require("../db/seeds/seed.js");
 const request = require("supertest");
+const endpointsData = require('../endpoints.json')
 
 beforeAll(() => seed(data));
 afterAll(() => db.end());
 
+describe('invalid enpoint - edge case', () => {
+    test('GET:404 "invalid endpoint" with a path of /api/topiks', () => {
+        return request(app)
+            .get('/api/topiks')
+            .expect(404)
+            .then(({body}) => {
+                expect(body.msg).toBe('invalid endpoint')
+            })
+    })
+})
 describe('/api/topics', () => {
     test("GET:200 sends an array of topics to the client", () => {
       return request(app)
         .get("/api/topics")
         .expect(200)
         .then(({body}) => {
-          expect(response.body.topics.length).toBe(3);
+          expect(body.topics.length).toBe(3);
           body.topics.forEach((topic) => {
             expect(typeof topic.description).toBe("string");
             expect(typeof topic.slug).toBe("string");
           });
         });
     });
-});
\ No newline at end of file
+});
+describe('/api', () => {
+    test("GET:200 sends an array of endpoints to the client", () => {
+        return request(app)
+          .get("/api")
+          .expect(200)
+          .then(({body}) => {
+            const {endpoints} = body
```

Comment:
Whilst this test is fine, it would be sufficient to check if what we receive equals the required in file